packages/webpack-dynamic-bundle-plugin/src/batch-dynamic-plugin.ts

```diff
@@ -0,0 +1,193 @@
+ */
+export default class DynamicBundlePlugin {
+  /**
+   * @param buildBatchEndpoint Helper to construct an asset URL from a serving
+   *                           prefix and a list of chunks ids.
+   */
+  constructor(private buildBatchEndpoint: (prefix: string, ids: number[]) => string) {
+  }
+
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.compilation.tap(DynamicBundlePlugin.name, (compilation) => {
+      let jsonpScriptHook = getWebpackJsonpHook(compilation);
+      jsonpScriptHook.tap(DynamicBundlePlugin.name, () => {
+        const {
+          crossOriginLoading,
+          chunkLoadTimeout,
+          jsonpScriptType
+        } = compilation.outputOptions;
+
+        const requireFn = '__webpack_require__';
+
+        const scriptNonce = `${requireFn}.nc`;
+        const activeBatch = `${requireFn}._ab`;
+        const publicPath = `${requireFn}.p`;
+
+        return Template.asString([
+          "var script;",
+          // DIFF: We create a batch entry and then have an if/else to either
+          // start a new batch or add it to an existing batch if one exists.
+          `var batchEntry = { id: chunkId, done: typeof loadingEnded === 'function' ? loadingEnded : ${loadingEnded4} };`,
+          `if (!${activeBatch}) {`,
+          Template.indent([
+            "script = document.createElement('script');",
+            "var batch = [batchEntry]",
+            `${activeBatch} = batch;`,
+            jsonpScriptType
+              ? `script.type = ${JSON.stringify(jsonpScriptType)};`
+              : "",
+            "script.charset = 'utf-8';",
+            `script.timeout = ${chunkLoadTimeout / 1000};`,
+            `if (${scriptNonce}) {`,
+            Template.indent(
+              `script.setAttribute("nonce", ${scriptNonce});`
+            ),
+            "}",
+            // DIFF: This delays the actual fetch by a microtick. In the original
+            // implementation of this hook, it's done immediately.
+            "Promise.resolve().then(function () {",
+            Template.indent([
+              `${activeBatch} = null;`,
+              // DIFF: In the original implementation, the script url is generated
+              // earlier and this is just `script.src = url;`.
+              `script.src = (${this.buildBatchEndpoint.toString()})(${publicPath}, batch.map(c => c.id));`,
+              crossOriginLoading
+                ? Template.asString([
+                  "if (script.src.indexOf(window.location.origin + '/') !== 0) {",
+                  Template.indent(
+                    `script.crossOrigin = ${JSON.stringify(crossOriginLoading)};`
+                  ),
+                  "}"
+                ])
+                : "",
+              "// create error before stack unwound to get useful stacktrace later",
+              "var error = new Error();",
+              "onScriptComplete = function (event) {",
+              Template.indent([
+                "// avoid mem leaks in IE.",
```

Comment:
So this means IE is not releasing the unreferenced onerror/onload functions?

Reply:
yeah, IE has (had?) some memory leaks around closures on DOM nodes. This was lifted from the original webpack template, not sure what support level they imply.